src/Apps/Settings/Routes/SavedSearchAlerts/SavedSearchAlertsApp.tsx

```diff
@@ -48,16 +54,20 @@ export const SavedSearchAlertsApp: React.FC<SavedSearchAlertsAppProps> = ({
     editAlertEntity,
     setEditAlertEntity,
   ] = useState<EditAlertEntity | null>(null)
+  const { relayEnvironment } = useSystemContext()
   const { sendToast } = useToasts()
   const { trackEvent } = useTracking()
   const [showDeleteModal, setShowDeleteModal] = useState(false)
   const [sort, setSort] = useState("CREATED_AT_DESC")
   const [loading, setLoading] = useState(false)
   const alerts = extractNodes(me.savedSearchesConnection)
   const isEditMode = editAlertEntity !== null
+  const { match } = useRouter()
+  const { jumpTo } = useJump()
 
   const closeEditForm = () => {
     setEditAlertEntity(null)
+    history.pushState({}, "", "/settings/alerts")
```

Comment:
we should use our router unless we're trying to (silently) update the url bar and not perform any actions. 

```tsx
router.push('/settings/alerts')
```

Reply:
This would actually be a nice framework addition since we do this in a number of places around force (and I had to add something similar to convos). 

Would you be willing to [update the `useRouter` hook](https://github.com/artsy/force/blob/main/src/System/Router/useRouter.tsx#L13) with two new functions -- `silentPush` and `silentReplace` which uses the code above?

Reply:
(Merged https://github.com/artsy/force/pull/13217 -- tyyy )